pkg/workceptor/workunitbase.go

```diff
@@ -402,7 +404,9 @@ func (bwu *BaseWorkUnit) MonitorLocalStatus() {
 	watcherErrors = make(chan error)
 
 	if bwu.watcher != nil {
+		watcherLock.Lock()
```

Comment:
Strange that we need this. 
Have you tried using the existing status file lock?